BEST_PRACTICES.md

```diff
@@ -0,0 +1,427 @@
+
+We practice the first one and we believe that you should too. It will make your work more clear and lucid. Thanks to that solution
+you can easily override mixins styles and while maintaining it will help you understand styles order faster. 
+
+Good practice:
+```scss
+.container {
+  @include p(2, 2, 2, 2);
+  @include font(h1, $align: right);
+  padding-left: 5px;
+}
+```
+
+Bad practice:
+```scss
+.container {
+  padding-left: 5px;
+  @include p(2, 2, 2, 2);
+  @include font(h1, $align: right);
+}
+```
+
+### Margins and paddings
+
+Using mixins for margins and paddings which we prepared will help you work faster and make your code cleaner if you ever want to maintain it.
+
+Good practice:
+```scss
+.container {
+  @include pl(2);
+  @include m(1);
+}
+```
+
+Bad practice:
+```scss
+.container {
+  padding-left: 20px;
+  margin: 10px;
+}
+```
+
+### Styling classes
+
+You should never style element. It's better to add unique class and style it. While styling elements you override previous style of given element.
+It may cause unwanted conflicts while using the same element in another place.
+
+Good practice:
+```scss
+.title {
+  @include font(h1);
+}
+```
+
+Bad practice:
+```scss
+h1 {
+  @include font(h1);
+}
+```
+
+### Use class instead of ID
+
+Using ids for styling is considered as one of bad practices. As you should already know classes are reusable and ids are unique. 
+Scss doesn't care if you use id or class but efficiency of your further work does.
+
+Good practice:
+```scss
+.title {
+  @include font(h1);
+}
+```
+
+Bad practice:
+```scss
+#title {
+  @include font(h1);
+}
+```
+
+### Unremovable empty classes
+
+<b>This rule we use only in no React projects.</b> 
+
+Sometimes in projects you need to get an element by it's class name using JavaScript. To make it easier to find, know
+ it's destination or just tell future developer to not remove it we use specific convention while naming it. To class's 
+ name we add `js-` prefix. Also what is important - this class should remain empty to be sure it is used only to find
+ specific element by this class.
+
+Good practice:
+```scss
+.js-specific-class {}
+```
+
+Bad practice:
+```scss
+.specific-class {
+  @include font(h1);
+  color: $red-error;
+}
+```
+
+### Global styles
+
+Sometimes we find a package with pretty awesome feature we want to add in our project but it doesn't fit our design. 
+To solve this problem good practice is to overwrite feature's styles with our own but in React it may be very tricky 
+(especially while using CSS Modules). That's why we recommend overwriting them in `:global` class. 
+
+Thanks to that during build step given class will be clear without any dynamically generated, unique, and mapped to the 
+correct styles additional text. They won't be scoped to particular templates but will be accessible from global scope.
+
+Good practice:
```

Comment:
plz add more examples, how to style external react-components which don't use css-modules: (it's good practice to wrap each `global` by normal classes)

```scss
.cell {
  @include flex($direction: column);
  @include p(0.5, 1);
  @include m(0.5, 0);

  :global {
    .Select.is-focused > .Select-control,
    .Select-control {
      border: none;
      border-bottom: 1px solid $color-gray-darker;
      background-color: transparent;
      border-radius: 0;
    }
  }
}
```

Reply:
how to style another react components which use css-modules, is it possible to change theirs styles?